salt/elife-xpub/scripts/setup-database.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -ex
+
+MAX_WAIT=20 # time in seconds
+DB_AVAILABLE_COMMAND="nc -z -w ${MAX_WAIT} ${PGHOST} ${PGPORT}"
+DC_COMMAND="docker-compose -f docker-compose.yml -f docker-compose.formula.yml"
+DB_CREATED_COMMAND="psql -c \"SELECT 'public.entities'::regclass\""
+
+SETUP_ARGS="--username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
+
+# Run the command to ensure the database is started
+$(${DC_COMMAND} start)
```

Comment:
Not sure why you need the `$()` around the command as you're not capturing the output. It doesn't suppress the `Starting ...` lines as they are on standard error, hence you would need `2>/dev/null`; but that may hide actual errors.

Besides:
- the `app` container should not be started here because in the case of a new environment there may be no database
- `start` only starts existing container that have previously been `stop`ped; unlike `up` and `down` which would create and delete containers.

The only container we should need here is `postgres` which is started separately in the formula (`{{ docker_compose }} up -d postgres`); that can be moved here in place of this if it makes things easier.